module-services/service-desktop/endpoints/filesystem/FS_Helper.cpp

```diff
@@ -329,28 +357,21 @@ namespace sdesktop::endpoints
             return ResponseContext{.status = http::Code::NotFound};
         }
 
-        std::vector<std::pair<std::string, int>> filesInDir;
-        for (const auto &entry : std::filesystem::directory_iterator{directory}) {
-            filesInDir.push_back(std::make_pair(entry.path(), entry.file_size()));
-        }
-
         json11::Json::array jsonArr;
-        jsonArr.reserve(filesInDir.size());
 
-        for (const auto &pathAndSize : filesInDir) {
-            jsonArr.push_back(
-                json11::Json::object{{json::fs::path, pathAndSize.first}, {json::fs::fileSize, pathAndSize.second}});
+        for (const auto &entry : std::filesystem::directory_iterator{directory}) {
+            jsonArr.push_back(parseFileEntry(entry));
         }
 
-        json11::Json::object response({{directory, jsonArr}});
+        json11::Json::object const response({{directory, jsonArr}});
         return ResponseContext{.status = http::Code::OK, .body = response};
     }
 
     auto FS_Helper::getFreeSpaceForUserFilesMiB() const -> unsigned long
     {
         const auto userDiskPath = purefs::dir::getUserDiskPath();
 
-        std::unique_ptr<struct statvfs> vfstat = std::make_unique<struct statvfs>();
+        auto vfstat = std::make_unique<struct statvfs>();
```

Comment:
How about simple allocation on stack? `struct vfs` should easily fit local stack :)

Reply:
I can check that, but we'll have to test thoroughly the whole feature :)